src/k8s-extension/azext_k8s_extension/custom.py

```diff
@@ -140,60 +141,77 @@ def create_k8s_extension(cmd, client, resource_group_name, cluster_name, name, c
     # Create identity, if required
     # We don't create the identity if we are in DF
     if create_identity and not is_dogfood_cluster(cmd):
-        extension_instance.identity, extension_instance.location = \
-            __create_identity(cmd, resource_group_name, cluster_name, cluster_type, cluster_rp)
+        identity_object, location = __create_identity(cmd, resource_group_name, cluster_name, cluster_type, cluster_rp)
+        if identity_object is not None and location is not None:
+            extension_instance.identity, extension_instance.location = identity_object, location
 
     # Try to create the resource
-    return client.create(resource_group_name, cluster_rp, cluster_type, cluster_name, name, extension_instance)
+    return sdk_no_wait(no_wait, client.begin_create, resource_group_name,
+                       cluster_rp, cluster_type, cluster_name, name, extension_instance)
 
 
 def list_k8s_extension(client, resource_group_name, cluster_name, cluster_type):
     cluster_rp = __get_cluster_rp(cluster_type)
     return client.list(resource_group_name, cluster_rp, cluster_type, cluster_name)
 
 
-def update_k8s_extension(client, resource_group_name, cluster_type, cluster_name, name,
-                         auto_upgrade_minor_version='', release_train='', version='', tags=None):
-
+def update_k8s_extension(cmd, client, resource_group_name, cluster_name, name, cluster_type,
+                         auto_upgrade_minor_version='', release_train='', version='',
+                         configuration_settings=None, configuration_protected_settings=None,
+                         configuration_settings_file=None, configuration_protected_settings_file=None,
+                         no_wait=False):
     """Patch an existing Extension Instance.
-
     """
+    msg = ('Updating properties in Configsettings or Configprotectedsettings may lead to undesirable state'
```

Comment:
There is a factory method implemented in default extension that also handles useres setting `confg.ignore_prompt`, use that instead